### DISCUSSION TITLE
workflow/first/

### DISCUSSION THREAD

**Discussion by giscus** [shane0/workflow#27] [General]
# workflow/first/

https://shanenull.com/workflow/first/

**Comment by shane0** [comment]
# challenge assumptions

## code

- lol do we need to rename the commands that already run fine using C or shell?
- https://rosettacode.org/wiki/Rosetta_Code

## keyboards

- I will make a page for this
- if you work at a computer the first principle is optimizing the input and that is you
- learn vim use ergonic programmable keyboards and keyboard layouts
- https://github.com/shane0/workflow/issues/26
- https://colemak.com/Learn
- https://www.vim.org/
- https://config.qmk.fm/#/handwired/dactyl/LAYOUT

this will sound harsh and negative but believe me if you use a standard keyboard you have stockholm syndrom and are torturing yourself without a clue what the good life on a keyboard is like lol

keyboards are one of the most moronic things

the first principal that keyboards violate is the keys should be under your fingers

the next is they should accomodate your hands

keyboard layouts were designed to solve a problem for the mechanical typewriter

they had bars underneath

does your computer have mechanical bars underneath preventing the keys to be positioned for the shape of your hand and fingers?

there are keyboards designed to match humans

they have keys under your fingers even matching the differences in length  and split apart since you have two arms...

they also are programmable with layers so you can move keys under your finger instead of moving your fingers to the keys

the thumbs are one reasons humans rose to the top of the food chain

and what do keyboards do?  waste all the mobility of thumbs on one huge spacebar

the caps lock key has a rockstar parking space but it is rarely used

delete / backspace is one of the most frequently used keys and is one of the furthest away from home position

along with escape

I can go on all day

not to mention vim keyboard shortcuts

one day I'll put a page here

**Reply by shane0** [reply]
https://shanenull.com/workflow/keyboards/

**Comment by shane0** [comment]
# quotes

- einstein said something like if he had an hour he'd spend 95 minutes defining the problem 5 minutes working on it

**Comment by shane0** [comment]
# types of reasoning

- inductive comprehension https://sutranet-org5.blogspot.com/2023/04/blog-post_24.html
- Cartesian
- Feynman method

**Reply by shane0** [reply]
What is comprehension? Comprehension is understanding the causes and consequences of a thing's changes, understanding how it came about and how it changed. When someone thinks about something and records the process, we can understand what they are saying after seeing the record. This is comprehension.

What is inductive ability? Inductive ability is the ability to discover patterns, connections, and commonalities among a large amount of unorganized information and data. This is inductive ability.

These two abilities are very important. Whether a person is good at thinking or not, the ability to understand and summarize accounts for a large proportion. In our daily lives, if we can use our understanding and induction abilities more often, we can gradually cultivate our thinking ability.
